solutions/1500-1599/1514-path-with-maximum-probability.md

```diff
@@ -0,0 +1,118 @@
+---
+description: >-
+  Author: @wingkwong |
+  https://leetcode.com/problems/path-with-maximum-probability/
+---
+
+# 1514 - Path with Maximum Probability (Medium)
+
+## Problem Link
+
+https://leetcode.com/problems/path-with-maximum-probability/
+
+## Problem Statement
+
+You are given an undirected weighted graph of n nodes (0-indexed), represented by an edge list where edges[i] = [a, b] is an undirected edge connecting the nodes a and b with a probability of success of traversing that edge succProb[i].
+
+Given two nodes start and end, find the path with the maximum probability of success to go from start to end and return its success probability.
+
+If there is no path from start to end, return 0. Your answer will be accepted if it differs from the correct answer by at most 1e-5.
+
+
+
+**Example 1:**
+
+```
+Input:  n = 3, edges = [[0,1],[1,2],[0,2]], succProb = [0.5,0.5,0.2], start = 0, end = 2
+Output: 0.25000
+Explanation: There are two paths from start to end, one having a probability of success = 0.2 and the other has 0.5 * 0.5 = 0.25.
+```
+
+**Example 2:**
+
+```
+Input: n = 3, edges = [[0,1],[1,2],[0,2]], succProb = [0.5,0.5,0.3], start = 0, end = 2
+Output: 0.30000
+```
+
+**Example 3:**  
+
+```
+Input: n = 3, edges = [[0,1]], succProb = [0.5], start = 0, end = 2  
+Output: 0.00000  
+Explanation: There is no path between 0 and 2.  
+```
+
+**Constraints:**
+
+* `2 <= n <= 10^4`
+* `0 <= start, end < n`
+* `start != end`  
+* `0 <= a, b < n`
+* `a != b`  
+* `0 <= succProb.length == edges.length <= 2*10^4`
+* `0 <= succProb[i] <= 1`
+* `There is at most one edge between every two nodes.`
+
+## Approach:  
+
+In standard djikstra algorithm, we will find the shortest path from start to end and update the "minimum distance" array accordingly for each "from -> to" pair.  
+In this question, we created a max heap for finding out maximum probability as asked in the problem statement.  
+The code for this approach is below : 
+
+
+## Code:  
+
+<Tabs>
+
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@YOUR_ALIAS"/>
+
+```cpp
```

Comment:
1. Stated in the checklist already. Don't write the comment on the same line with your code.

```
Make sure you've explained your code well (write comments above each line - not inline)
```

2. add some spaces. e.g.

```
for(int i=0;i<edges.size();i++){
```

to

```
for (int i = 0; i < edges.size(); i++) {
```

Reply:
Same for the second point. check `if`, `for` etc.. also for the brackets. please double check all.

e.g. prefer

```cpp
for (int i = 0; i < total_edges; i++) {
    graph[edges[i][0]].push_back({edges[i][1], succProb[i]});
    graph[edges[i][1]].push_back({edges[i][0], succProb[i]});
}
```

rather than

```cpp
for(int i = 0; i < total_edges; i++)
{
    graph[edges[i][0]].push_back({edges[i][1], succProb[i]});
    graph[edges[i][1]].push_back({edges[i][0], succProb[i]});
}
```